confidential-data-hub/storage/src/volume_type/mod.rs

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2023 Intel
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+pub mod alibaba_cloud_oss;
+
+use anyhow::anyhow;
+use self::alibaba_cloud_oss::oss::Oss;
+use crate::{Error, Result};
+
+#[derive(PartialEq, Clone, Debug)]
+pub struct Storage {
+    pub driver: String,
+    pub driver_options: Vec<String>,
+    pub source: String,
+    pub fstype: String,
+    pub options: Vec<String>,
+    pub mount_point: String,
+}
+
+impl Storage {
+    pub async fn mount(&self) -> Result<String> {
+        let volumetype: String;
+        for driver_option in &self.driver_options {
+            if driver_option.starts_with("volumetype=") {
+                volumetype = driver_option
+                    .strip_prefix("volumetype=")
+                    .ok_or(anyhow!("strip_prefix \"volumetype=\" failed"))
+                    .map_err(|e| {
+                        Error::SecureMountFailed(format!(
+                            "strip_prefix \"volumetype=\" failed: {e}"
+                        ))
+                    })?
+                    .to_string();
+
+                match volumetype.as_str() {
+                    "alibaba-cloud-oss" => {
```

Comment:
Maybe we should have a feature to enable/disable particular storage vendors?

Reply:
fixed, thanks